packages/@guidesmiths/cuckoojs-schematics/src/helm/helm-strategies/ingress-controller/ingress.strategy.factory.ts

```diff
@@ -1,14 +1,15 @@
-import {HelmStrategyFactory} from '../helm.strategy.factory';
-import {TraefikIngressControllerStrategy} from './traefik.Ingress.strategy';
-import {GenericIngressControllerStrategy} from './generic.Ingress.strategy';
-import {type IngressStrategy} from './ingress.strategy';
+import { HelmStrategyFactory } from '../helm.strategy.factory';
+import { TraefikIngressControllerStrategy } from './traefik.Ingress.strategy';
+import { GenericIngressControllerStrategy } from './generic.Ingress.strategy';
+import { type IngressStrategy } from './ingress.strategy';
+import { IngressTypes } from './ingressTypes.enum';
 
 export class IngressStrategyFactory extends HelmStrategyFactory<IngressStrategy> {
-	create(type: string): IngressStrategy {
-		const switcher: Record<string, () => IngressStrategy> = {
-			generic: () => new GenericIngressControllerStrategy(),
-			traefik: () => new TraefikIngressControllerStrategy(),
-		};
-		return switcher[type]?.() || switcher.generic();
-	}
+  create(type: string): IngressStrategy {
```

Comment:
Would it make sense to describe the type argument as  `typeof IngressTypes[keyof typeof IngressTypes]`?

Reply:
We can probably type this, yes. Let me have a look